integration/test-mb/test-amqp/src/test/java/org/wso2/ei/mb/test/amqp/AcknowledgementsTestCase.java

```diff
@@ -0,0 +1,308 @@
+     * 3. Receiver receives {@link #EXPECTED_COUNT}
+     * 4. Check whether all messages received.
+     *
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     * @throws InterruptedException
+     */
+    @Test
+    public void autoAcknowledgementsTestCase()
+            throws JMSException, NamingException, IOException, InterruptedException {
+
+        // Creating a JMS consumer
+        QueueReceiver queueReceiver = new QueueReceiver("autoAckTestQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueReceiver.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        // Creating a JMS publisher
+        QueueSender queueSender = new QueueSender("autoAckTestQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueSender.sendMessages(SEND_COUNT, "text message");
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        int receivedMessageCount = queueReceiver.receivedMessageCount();
+
+        // close queue sender.
+        queueSender.closeSender();
+
+        // close queue receiver .
+        queueReceiver.closeReceiver();
+
+        // Evaluating results
+        Assert.assertEquals(receivedMessageCount, EXPECTED_COUNT, "Total number of sent and received messages" +
+                " are not equal");
+    }
+
+    /**
+     * In this method we drop receiving client and connect it again and tries to get messages from MB.
+     * 1. Start a queue receiver in auto acknowledge mode.
+     * 2. Publishers sends {@link #SEND_COUNT} number of messages.
+     * 3. First receiver will read up to first 500 messages.
+     * 4. Close up the receiver.
+     * 5. Start a second queue receiver in auto acknowledge mode.
+     * 6. Second receiver will read up 500 messages.
+     * 7. Check whether total received messages were equal to {@link #EXPECTED_COUNT}.
+     *
+     * @throws InterruptedException
+     * @throws JMSException
+     * @throws NamingException
+     * @throws IOException
+     */
+    @Test
+    public void autoAcknowledgementsDropReceiverTestCase() throws InterruptedException, JMSException, NamingException,
+            IOException {
+
+        // Creating a initial JMS consumer with autoack mode
+        QueueReceiver queueReceiver = new QueueReceiver("autoAckDropReceiverTestQueue",
+                JMSAcknowledgeMode.AUTO_ACKNOWLEDGE, clientConfigPropertiesMap);
+        queueReceiver.registerSubscriber();
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        // Creating a JMS publisher
+        QueueSender queueSender = new QueueSender("autoAckDropReceiverTestQueue", JMSAcknowledgeMode.AUTO_ACKNOWLEDGE,
+                clientConfigPropertiesMap);
+        queueSender.sendMessages(SEND_COUNT, "text message");
+        int receivedMessageCountSub1 = 0;
+
+        TimeUnit.SECONDS.sleep(50L);
+
+        // Wait until 500 messages are received by first consumer client.
+        while (true) {
```

Comment:
Yes can add a wait for some feasible time and if all messages are not received within the interval need to break the loop there also.